WeakAurasOptions/OptionsFrames/IconPicker.lua

```diff
@@ -33,7 +33,13 @@ local function ConstructIconPicker(frame)
     local distances = {};
     local names = {};
 
-    subname = tonumber(subname) and GetSpellInfo(tonumber(subname)) or subname;
+    if (tonumber(subname)) then
+      local spellId = tonumber(subname);
+      -- Work around special numbers such as inf and nan
+      if (abs(spellId) < math.huge and spellId == spellId) then
```

Comment:
As I recall, blizzard has a very nonstandard approach to NaN, and guarding against it via spellId == spellId won’t work. Has this changed?

Reply:
Ah no it didn't change. I did change that after testing, but apparently didn't commit it.